M/MPFR/common.jl

```diff
@@ -0,0 +1,72 @@
+using BinaryBuilder
+using BinaryBuilderBase: sanitize
+using Pkg
+
+# Collection of sources required to build MPFR
+function mpfr_sources(version::VersionNumber; kwargs...)
+    mpfr_version_sources = Dict(
+        v"4.1.1" => [
+            ArchiveSource("https://www.mpfr.org/mpfr-$(version)/mpfr-$(version).tar.xz",
+                          "ffd195bd567dbaffc3b98b23fd00aad0537680c9896171e44fe3ff79e28ac33d"),
+        ],
+        v"4.2.0" => [
+            ArchiveSource("https://www.mpfr.org/mpfr-$(version)/mpfr-$(version).tar.xz",
+                          "06a378df13501248c1b2db5aa977a2c8126ae849a9d9b7be2546fb4a9c26d993"),
+        ],
+        v"4.2.1" => [
+            ArchiveSource("https://www.mpfr.org/mpfr-$(version)/mpfr-$(version).tar.xz",
+                          "277807353a6726978996945af13e52829e3abd7a9a5b7fb2793894e18f1fcbb2"),
+        ],
+        v"4.2.2" => [
+            ArchiveSource("https://www.mpfr.org/mpfr-$(version)/mpfr-$(version).tar.xz",
+                          "b67ba0383ef7e8a8563734e2e889ef5ec3c3b898a01d00fa0a6869ad81c6ce01"),
+        ],
+    )
+    return [
+        mpfr_version_sources[version]...,
+    ]
+end
+
+# Bash recipe for building across all platforms
+function mpfr_script(; kwargs...)
+    script = raw"""
+    if [[ ${bb_full_target} == *-sanitize+memory* ]]; then
+        # Install msan runtime (for clang)
+        cp -rL ${libdir}/linux/* /opt/x86_64-linux-musl/lib/clang/*/lib/linux/
+    fi
+
+    cd $WORKSPACE/srcdir/mpfr-*
+    ./configure --prefix=${prefix} --build=${MACHTYPE} --host=${target} --enable-shared --disable-static --with-gmp=${prefix} --enable-thread-safe --enable-shared-cache --disable-float128 --disable-decimal-float
+    make -j${nproc}
+    make install
+
+    # On Windows, make sure non-versioned filename exists...
+    if [[ ${target} == *mingw* ]]; then
+        cp -v ${libdir}/libmpfr-*.dll ${libdir}/libmpfr.dll
+    fi
+    """
+    return script
+end
+
+function mpfr_platforms(; experimental::Bool=true, kwargs...)
+    platforms = supported_platforms(; experimental)
```

Comment:
```suggestion
    platforms = supported_platforms()
```
Not needed.

Reply:
I know, but allowing to pass the kwarg seems like future-proofing for me (this is taken from the OpenBLAS `common.jl`). If you want to, I can get rid of this, but then I would also remove the kwarg in the line above.

Reply:
`experimental` was only used for the Apple aarch64 M1 bringup, and mainly because pre-1.6 versions choked on the idea of aarch64 darwin. The riscv platform that was just rolled out didn't use `experimental`, and I doubt it will be used anytime in the future since Julia 1.6+ gracefully supports unknown platforms: https://github.com/JuliaPackaging/Yggdrasil/pull/5057#discussion_r902929036.

And yep, remove the kwarg as well.